spec/lib/theme_settings_manager_spec.rb

```diff
@@ -161,12 +161,30 @@ def find_by_name(name)
       ).to be_truthy
     end
 
-    it "returns the CDN URL" do
-      upload_setting = find_by_name(:upload_setting)
-      upload_setting.value = upload.url
-      theme.reload
-
-      expect(upload_setting.value).to eq(Discourse.store.cdn_url(upload.url))
+    describe "#value" do
+      context "when it's changed to a custom upload" do
+        it "returns CDN URL" do
+          upload_setting = find_by_name(:upload_setting)
+          upload_setting.value = upload.url
+          theme.reload
+
+          expect(upload_setting.value).to eq(Discourse.store.cdn_url(upload.url))
+        end
+      end
+
+      context "when there's a default upload" do
+        it "returns CDN URL" do
+          theme.set_field(
+            target: :common,
+            name: "default-upload",
+            type: :theme_upload_var,
+            upload_id: upload.id,
+          )
+          theme.save!
+          upload_setting = find_by_name(:upload_setting)
+          expect(upload_setting.value).to eq(Discourse.store.cdn_url(upload.url))
+        end
+      end
```

Comment:
The tests are a bit interesting. Should you be using `default_upload_id` somewhere?

Reply:
`default_upload_id` is just a private method defined way at the end. It queries the `theme_fields` table for an theme field with a name that matches the setting's default value. In this test, I simply create a theme field that the `default_upload_id` method can find.

Reply:
Got it, it's a derived value which you've set up 👌

Reply:
Yep, exactly!